ui/app/src/render-app.tsx

```diff
@@ -36,12 +38,14 @@ export function renderApp(container: Element | null) {
     <React.StrictMode>
       <BrowserRouter>
         <QueryClientProvider client={queryClient}>
-          <DarkModeContextProvider>
-            <SnackbarProvider anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}>
-              <CssBaseline />
-              <Router />
-            </SnackbarProvider>
-          </DarkModeContextProvider>
+          <QueryParamProvider adapter={ReactRouter6Adapter}>
```

Comment:
I'm wondering if it's better to add `QueryParamProvider` into the `DashboardView` component in the `dashboard` package so all the providers are self contained. I suppose this way is more consistent with `QueryClientProvider` as well. But maybe both should go in there and we can pass optional params:

```tsx
<DashboardView reactQueryClient={queryClient} urlAdapter={ReactRouter6Adapter}>
...
</DashboardView>
```

Maybe in a follow up we can think about this more.

Reply:
Yeah that sounds good for follow up! I did try this briefly, but it was going to require moving DashboardProvider, TimeRangeProvider, etc. down a level from that ViewDashboard since they rely on query params to [populate initialTimeRange](https://github.com/perses/perses/blob/33a590cee37a10f1cee92d98348df6417ea268e7/ui/dashboards/src/views/ViewDashboard/ViewDashboard.tsx#L39) (and we aren't importing that ViewDashboard internally anymore anyways since we've rebuilt DashboardToolbar).

Reply:
To add to comment above, I wanted to avoid adding QueryParamProvider and QueryClientProvider in dashboards package's ViewDashboard (aka DashboardView) since that would mean duplicating those two providers since they're also needed in the Perses app.